src/bulk.rs

```diff
@@ -795,17 +831,25 @@ impl Rows {
             return Ok(());
         }
 
-        // Process all rows in the buffer at once for better performance
-        let rows = std::mem::take(&mut self.row_buffer);
-        self.builder.add_rows(rows)?;
+        if let RowsData::Builder(ref mut builder) = self.data {
+            // Process all rows in the buffer at once for better performance
+            let rows = std::mem::take(&mut self.row_buffer);
+            builder.add_rows(rows)?;
+        } else {
+            // This indicates a logic error - flushing a buffer without a builder
+            ensure!(self.row_buffer.is_empty(), error::UnflushedRowsSnafu,);
```

Comment:
The `row_buffer` must be empty because it is checked at the beginning of this method.

Reply:
Correct. Sorry for this oversight.